RHDEVDOCS-6460: Content creation for handling YAML parsing errors in PAC (#94302)
openshift/openshift-docs: modules/op-monitoring-pipeline-run-status-using-pipelines-as-code.adoc

.Failures
If a namespace is matched to a `Repository` custom resource definition (CRD), {pac} emits its failure log messages in the Kubernetes events inside the namespace.

[discrete]
.YAML parting error messages in pull request comments

> {pac} detects and reports YAML errors in PipelineRun definitions within the .tekton directory. If a formatting issue is found, it logs the error in the user namespace events log and the {pac} controller log, and comments on the pull request for GitHub, Gitea, and GitLab. These errors do not halt the execution of valid pipeline runs because the system continues to process correctly formatted and matched pipeline runs, preventing a single formatting problem from disrupting the entire automation workflow. When an invalid PipelineRun YAML is found in a pull request (PR) event, {pac} creates a comment on the PR describing the error. If the PR is updated and new YAML errors are detected, the comment is updated with the latest information. If no new errors are found, the existing comment remains unchanged and is not deleted.

Reviewer: ```suggestion
{pac} detects and reports YAML errors in PipelineRun definitions within the .tekton directory. When an invalid PipelineRun YAML is found in a pull request (PR) event, {pac} creates a comment on the PR describing the error and halt the execution of other correctly formatted pipeline runs. If the PR is updated and new YAML errors are detected, the comment is updated with the latest information. If no new errors are found, the existing comment remains unchanged and is not deleted. If a validation issue is found, it logs the error in the user namespace events log and the {pac} controller log. These errors do not halt the execution of valid pipeline runs because the system continues to process validated and matched pipeline runs, preventing a single validation problem from disrupting the entire automation workflow.
```

.Failures
If a namespace is matched to a `Repository` custom resource definition (CRD), {pac} emits its failure log messages in the Kubernetes events inside the namespace.

[discrete]
.YAML parting error messages in pull request comments
{pac} can detect and report errors in the YAML definition of a `PipelineRun`. When a YAML formatting issue is encountered in the repository, the system logs the error in the user namespace events log and the {pac} controller log.

> These errors do not halt the execution of valid pipeline runs because the system continues to process correctly formatted and matched pipeline runs, preventing a single formatting problem from disrupting the entire automation workflow.

Author: Any suggestions for improving the content @zakisk @chmouel???

> These errors do not halt the execution of valid pipeline runs because the system continues to process correctly formatted and matched pipeline runs, preventing a single formatting problem from disrupting the entire automation workflow.

Reviewer: ```suggestion
```

Author: Fixed.

> When a pull request (PR) triggers the event, {pac} creates a comment on the PR that describes the error. If the PR is updated and new YAML errors are detected, the comment is updated with the latest information. If no new errors are found, the existing comment remains unchanged and is not deleted.

Author: Fixed.

> When a pull request (PR) with a YAML error triggers the event, {pac} creates a comment on the PR that describes the error. If the PR is updated and new YAML errors are detected, the comment is updated with the latest information. If no new errors are found, the existing comment remains unchanged and is not deleted.

Reviewer: ```suggestion
When an invalid PipelineRun YAML found in a pull request (PR) event, {pac} creates a comment on the PR that describes the error. If the PR is updated and new YAML errors are detected, the comment is updated with the latest information. If no new errors are found, the existing comment remains unchanged and is not deleted.
```

Reviewer: @Dhruv-Soni11 YAML error doesn't trigger an event rather PipelineRun YAMLs are validated in on pull request event.
